### ISSUE TITLE
Manual search for subtitles by text

### ISSUE THREAD

**Issue by zennethos** [iina/iina#4773]
have an option to manually search for subtitles using entry like "bojack horseman s03e07"

currently it just automatically takes the file name and searches for it but often times when im playing from a url the file name is random and not the title of the movie or tv show i'm watching.

it'll be helpful if we get a bit of control over the searching of subtitles.

vlc has this option and its pretty great.

**Comment by low-batt** [comment]
I've labeled this as a feature request for the ability to override the search string when searching for subtitles.

Note that when streaming from a URL, IINA will only use the URL as the search string if the title of the video is not available. Playing the following video: 
https://www.youtube.com/watch?v=tO01J-M3g0U

The log file shows IINA was able to obtain the title from YouTube and used that when searching for subtitles:
```text
21:36:07.827 [opensub][d] Searching for subtitles of movies matching 'The World in HDR in 4K (ULTRA HD)'
```

The `mpv` log shows the property IINA reads to obtain the title being initialized:
```text
[  39.952][v][cplayer] Set property: file-local-options/force-media-title="The World in HDR in 4K (ULTRA HD)" -> 1
```

**Comment by rasitayaz** [comment]
I agree that this would be a very useful feature. Additionally, IINA should prefer metadata information over the filename if/when available.

**Comment by AB-boi** [comment]
> Note that when streaming from a URL, IINA will only use the URL as the search string if the title of the video is not available.

It would be great to have a way to search since the URL may not always have the correct title. A manual search is a must need feature!

**Comment by low-batt** [comment]
For some features IINA is in the process of transitioning from a builtin implementation to a plugin implementation. As of IINA 1.4.0 by default IINA has transitioned to officially supported plugins for accessing [Online Media](https://github.com/iina/plugin-online-media) and for downloading subtitles from [OpenSubtitles](https://github.com/iina/plugin-opensub). As of now, if you disable the plugins the old builtin implementation is still available.

One advantage of using plugins is that updates to the implementation do not have to wait for the next IINA release. Follow these instructions:
- Start IINA
- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel click on `Plugins`
- On the right side click on `OpenSubtitles`
- Confirm the version shown is 0.9.5
- If not, click on `Check for Updates` and install the latest version

![Image](https://github.com/user-attachments/assets/08cdc4db-ed8d-4f98-9bfa-fbae737e486f)

When you want to specify the name to search subtitles for, click on the `Show Sidebar` menu item in the `OpenSubtitles` section of the `Plugin` menu:

![Image](https://github.com/user-attachments/assets/32fdd358-017d-4fc8-bbf2-2836b2671723)

The plugin will open a sidebar that contains a text field that allows you to specify the text to search for:

![Image](https://github.com/user-attachments/assets/ea7bbac1-559d-4b3a-b0f8-e3e5daf8d001)

Fill in the text field and click on `Go`. Clicking the `…` button exposes additional controls for filtering the selection. Notice that if many possible subtitles are returned there may be multiple pages. Select one of the subtitles and click on the download button:

![Image](https://github.com/user-attachments/assets/210f2deb-a619-4367-afa7-6a7d130ba415)

The subtitles will be downloaded and selected:

![Image](https://github.com/user-attachments/assets/090f2ac1-4b8d-4d50-b8d2-2361e896211a)

**Comment by AB-boi** [comment]
@low-batt this plugin does not show a search box at times:

https://github.com/user-attachments/assets/2ea1cc0a-805e-4cef-97bb-34fdf9411db8

Also, does this plugin need you to be logged in?

**Comment by low-batt** [comment]
First, on this:
> Also, does this plugin need you to be logged in?

No, but it is more likely to work if you are logged in. This is more about [Open Subtitles](https://www.opensubtitles.com/) than the plugin.

From the Open Subtitles Help Center post [API - Getting Started](https://opensubtitles.tawk.help/article/getting-started):
> ## Step 6: Understand Quotas and Limits
>
> While there is no limit on searching for subtitles, downloading is restricted:
>
> -    Anonymous Users: 5 downloads per 24 hours per IP.
> -    Authenticated Users: Depending on their rank, users can download more subtitles—from 10 for regular users to 1000 for VIP users.

From a high level perspective use of Open Subtitles can be broken down into:
- Anonymous user (identified by IP address)
- Authenticated free user
- Authenticated paid VIP user

One important advantage of the VIP account is that when you log in you are redirected to Open Subtitles servers that are dedicated to serving VIP users. Anonymous and free users are serviced by the public Open Subtitles servers. When the servers get overloaded they do not always respond appropriately to clients. At least that was the case when I was working on IINA's builtin code. In a quick search I found the post [Site is slow or not working](https://forum.opensubtitles.com/t/site-is-slow-or-not-working/5105) from Feb 2<sup>nd</sup>, so server overload may still be a problem.

As anonymous users are identified by their IP address, if you are on a shared network behind a firewall, other users may have already used up the download limit. So it is desirable to sign up for a free account. That will also give you a higher download limit. More details on accounts can be found Open Subtitles Help Center post [OpenSubtitles User Levels: From Basic to VIP](https://opensubtitles.tawk.help/article/user-levels-basic-vip).

In the example I showed above I was not logged in.

On this:
> this plugin does not show a search box at times:

I've opened issue https://github.com/iina/plugin-opensub/issues/4 against the plugin for that.

I tried repeating the example I posted above and after an OSC message indicating it was downloading subtitles **nothing happened**.  Checking the IINA log file showed OpenSubtitles refused to download the file and instead returned this error:

> You have downloaded your allowed 100 subtitles for 24h.Your quota will be renewed in 05 hours and 33 minutes (2026-02-13 23:59:59 UTC) ts=1771007207 

News to me! Seems my ISP has deployed [CGNAT](https://en.wikipedia.org/wiki/Carrier-grade_NAT) and I am now sharing an IP address with many other customers.

This is what the failure looked like in the IINA log:

<details>
<summary><b>IINA log:</b></summary>

```text
13:26:46.917 [player0 - OpenSubtitles][e] Error requesting /download
13:26:46.918 [player0 - OpenSubtitles][e] {
  "status": 406,
  "data": {
    "uk": "app_ud_82678",
    "uid": 2,
    "ts": 1771007207,
    "message": "You have downloaded your allowed 100 subtitles for 24h.Your quota will be renewed in 05 hours and 33 minutes (2026-02-13 23:59:59 UTC) ts=1771007207 ",
    "reset_time": "05 hours and 33 minutes",
    "remaining": -1,
    "requests": 101,
    "reset_time_utc": "2026-02-13T23:59:59.000Z"
  },
  "statusText": "Not Acceptable",
  "headers": {
    "last-modified": "Fri, 13 Feb 2026 18:26:47 GMT",
    "pragma": "no-cache",
    "content-type": "application/json; charset=utf-8",
    "cache-control": "max-age=0, private, must-revalidate, s-maxage=600",
    "expires": "0"
  },
  "config": {
    "timeout": 0,
    "xsrfHeaderName": "X-XSRF-TOKEN",
    "headers": {
      "User-Agent": "IINAPlugin v0.0.1",
      "Api-Key": "********",
      "Content-Type": "application/json",
      "Accept": "*/*"
    },
    "maxContentLength": -1,
    "url": "https://api.opensubtitles.com/api/v1/download",
    "params": {},
    "transformResponse": [
      null
    ],
    "adapter": [
      "xhr",
      "http"
    ],
    "transformRequest": [
      null
    ],
    "data": "{\"file_id\":4638179}",
    "maxBodyLength": -1,
    "env": {},
    "transitional": {
      "clarifyTimeoutError": false,
      "silentJSONParsing": true,
      "forcedJSONParsing": true
    },
    "xsrfCookieName": "XSRF-TOKEN",
    "method": "post"
  },
  "request": {}
}
```
</details>

I will be entering an issue against the plugin for not displaying an OSD message indicating the download failed.

The failure also does not display in the log shown when you click on the `Log` button:

![Image](https://github.com/user-attachments/assets/88368b8c-5e98-46a2-8427-2cf52b4fa335)

With ISPs starting to use [CGNAT](https://en.wikipedia.org/wiki/Carrier-grade_NAT) it is more important to use an OpenSubtitles account. I clicked on `Log in` and logged into my OpenSubtitles account:

![Image](https://github.com/user-attachments/assets/60c149f6-6ccb-4387-8553-255e1bb3775f)

And was able to download subtitles.

At least when logged in you can click on your username and see information about your quota:

![Image](https://github.com/user-attachments/assets/e002b14a-dac9-4681-96e0-6c0628b42bce)

As for why the plugin is failing to display the sidebar…

I have worked on the old builtin code, but the plugin was developed by another developer. My only guess on why the sidebar is malfunctioning is that maybe a thread is hung up waiting for a response from OpenSubtitles. Will have to see what the response is to issue https://github.com/iina/plugin-opensub/issues/4.

**Comment by AB-boi** [comment]
Awesome! Thanks a lot for the detailed answer and for opening the issue! Good day to you!
